slices.go

```diff
@@ -66,10 +66,23 @@ func Shuffle[T any](s []T) []T {
 
 // Reverse reverses the order of s.
 func Reverse[T any](s []T) []T {
-	for i := len(s)/2-1; i >= 0; i-- {
-		opp := len(s)-1-i
+	for i := len(s)/2 - 1; i >= 0; i-- {
+		opp := len(s) - 1 - i
 		s[i], s[opp] = s[opp], s[i]
 	}
 
 	return s
 }
+
+// Unique removes duplicate values in a slice
+func Unique[T comparable](src []T) []T {
+	var result []T
+	var elemMap = make(map[T]bool)
+	for i := range src {
+		if !elemMap[src[i]] {
+			result = append(result, src[i])
+			elemMap[src[i]] = true
```

Comment:
Would using empty struct be more efficient?

```suggestion
	var elemMap = make(map[T]struct{})
	for i := range src {
		if _, ok := elemMap[src[i]]; !ok {
			result = append(result, src[i])
			elemMap[src[i]] = struct{}{}
```

Reply:
copy that. I made those changes and pushed.